coa/pkg/apis/v1alpha2/providers/states/redisstate/redisstate.go

```diff
@@ -169,15 +207,22 @@ func (r *RedisStateProvider) Upsert(ctx context.Context, entry states.UpsertRequ
 		}
 		oldEntryDict["status"] = oldStatusDict
 		body, _ = json.Marshal(oldEntryDict)
-		_, err = r.Client.HSet(r.Ctx, key, "values", string(body)).Result()
+		if etag == "" {
+			etag = "0"
+		}
+		err = r.Client.Do(r.Ctx, "EVAL", setDefaultQuery, 1, key, etag, string(body), 1).Err()
```

Comment:
Should the last parameter be firstWrite?

Reply:
I really want to remove UpdateStatusOnly option because the caller should get the previous state first, apply the update and then upsert with etag.
Here the upsert operation is based on the previous get result so the firstWrite should always be 1.

Reply:
I would prefer we still keep UpdateStatusOnly since like K8S state provider, status are sub resources. But seems we should not always default to useFirstWriteWins or lastWriteWins when updating status. As state provider, it is better we align client's input. And for all clients, they need to handle get state at first before composing new state.